manifests/config/postgres.pp

```diff
@@ -0,0 +1,29 @@
+# == Define: xylem::config::postgres
+#
+# [*postgres_host*]
+#   Host to create databases on.
+#
+# [*postgres_user*]
+#   User to create databases with.
+#
+# [*postgres_secret*]
+#   Secret key for storing generated credentials.
+#
+# [*postgres_password*]
+#   Optional password for postgres user.
+#
+class xylem::config::postgres(
+  $postgres_host,
+  $postgres_user,
+  $postgres_secret,
+  $postgres_password = undef,
+) {
+  if $postgres_host == undef { fail('postgres_host must be provided') }
+  if $postgres_user == undef { fail('postgres_user must be provided') }
+  if $postgres_secret == undef { fail('postgres_secret must be provided') }
```

Comment:
We still need them, because it's possible to pass `postgres_host => undef`. (That's what our tests do, and no error is raised if we don't have the check.)